docs/production/troubleshooting.md

```diff
@@ -256,7 +256,7 @@ Database monitoring:
 - `check_postgres`: General checks for database health.
 - `check_postgresql_backup`: Checks status of PostgreSQL backups.
 - `check_postgresql_replication_lag`: Checks whether PostgreSQL streaming
-  replication is up to date.
+  replication is up-to-date.
```

Comment:
See https://community.languagetool.org/rule/show/UP_TO_DATE_HYPHEN?lang=en&subId=1.

Reply:
This rule is, sadly, wrong. The sentence we have is correct.

It's true that "up to date replication" would be wrong, and should be "up-to-date replication". But when the phrase "up to date" isn't directly attached to the thing it's modifying, it's just a phrase with no hyphens: "replication is up to date".

Awkwardly, the example in that docs page is of precisely the case where the rule is wrong, and turns a correct spelling into an incorrect one:
> The software is up to date.
> Correction suggestion: up-to-date

The rule would do the right thing on a different example: *"This is up to date software" is wrong, and should be "This is up-to-date software".

Reply:
Oof, and the article they link to (which is on their own blog) is full of wrong examples mixed in with right ones. … I'm going to not even link to it, because I can't recommend even looking at it -- it carries a risk of making one's writing worse. Seeing that post, I do not believe that the author is a native speaker of English, nor someone who has spent a large part of their life reading books and newspapers and such in English. They can write clearly enough, but for them to go publishing advice about tricky nuances of English grammar is negligent.

In amongst all the things they get wrong, they actually have a pair of examples that correctly demonstrate why this rule is wrong! They correctly identify both of these sentences as correct:

> George Orwell was a well-known author.
> George Orwell was well known for his novels.

Saying "replication is up to date" behaves just the same way as "George Orwell was well known".

Reply:
Should we raise an issue in https://github.com/languagetool-org/languagetool?